src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeServiceClient.java

```diff
@@ -0,0 +1,222 @@
+import static net.snowflake.ingest.connection.ServiceResponseHandler.ApiName.STREAMING_REGISTER_BLOB;
+import static net.snowflake.ingest.streaming.internal.StreamingIngestUtils.executeWithRetries;
+import static net.snowflake.ingest.utils.Constants.CHANNEL_CONFIGURE_ENDPOINT;
+import static net.snowflake.ingest.utils.Constants.CHANNEL_STATUS_ENDPOINT;
+import static net.snowflake.ingest.utils.Constants.CLIENT_CONFIGURE_ENDPOINT;
+import static net.snowflake.ingest.utils.Constants.DROP_CHANNEL_ENDPOINT;
+import static net.snowflake.ingest.utils.Constants.OPEN_CHANNEL_ENDPOINT;
+import static net.snowflake.ingest.utils.Constants.REGISTER_BLOB_ENDPOINT;
+import static net.snowflake.ingest.utils.Constants.RESPONSE_SUCCESS;
+
+import java.io.IOException;
+import java.util.stream.Collectors;
+import net.snowflake.client.jdbc.internal.apache.http.impl.client.CloseableHttpClient;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.connection.RequestBuilder;
+import net.snowflake.ingest.connection.ServiceResponseHandler;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.Logging;
+import net.snowflake.ingest.utils.SFException;
+
+/**
+ * The SnowflakeServiceClient class is responsible for making API requests to the Snowflake service.
+ */
+class SnowflakeServiceClient {
+  private static final Logging logger = new Logging(SnowflakeServiceClient.class);
+
+  // HTTP client used for making requests
+  private final CloseableHttpClient httpClient;
+
+  // Request builder for building streaming API request
+  private final RequestBuilder requestBuilder;
+
+  /**
+   * Default constructor
+   *
+   * @param httpClient the HTTP client used for making requests
+   * @param requestBuilder the request builder for building streaming API requests
+   */
+  SnowflakeServiceClient(CloseableHttpClient httpClient, RequestBuilder requestBuilder) {
+    this.httpClient = httpClient;
+    this.requestBuilder = requestBuilder;
+  }
+
+  /**
+   * Configures the client given a {@link ClientConfigureRequest}.
+   *
+   * @param request the client configuration request
+   * @return the response from the configuration request
+   */
+  ClientConfigureResponse clientConfigure(ClientConfigureRequest request)
+      throws IngestResponseException, IOException {
+    ClientConfigureResponse response =
+        executeApiRequestWithRetries(
+            ClientConfigureResponse.class,
+            request,
+            CLIENT_CONFIGURE_ENDPOINT,
+            "client configure",
+            STREAMING_CLIENT_CONFIGURE);
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug(
+          "Client configure request failed, request={}, message={}",
+          request.getStringForLogging(),
+          response.getMessage());
+      throw new SFException(ErrorCode.CLIENT_CONFIGURE_FAILURE, response.getMessage());
```

Comment:
why do we need both the log and the exception?

Reply:
ok, looks like we also did both before, but the log has more info, like table/channel name, could we make sure to keep the original logic? My assumption is that this is a pure refactoring so let's not change any logic, thanks!

Reply:
All information including table/channel name are capsulated in `getStringForLogging()`.